Python/pystrcmp.c

```diff
@@ -6,21 +6,24 @@
 int
 PyOS_mystrnicmp(const char *s1, const char *s2, Py_ssize_t size)
 {
+    const unsigned char *p1, *p2;
     if (size == 0)
         return 0;
-    while ((--size > 0) &&
-           (tolower((unsigned)*s1) == tolower((unsigned)*s2))) {
-        if (!*s1++ || !*s2++)
+    p1 = (void *)s1;
```

Comment:
Casting pointer to constant value to a pointer to non-constant value can emit a compiler warning. It is better to preserve constness: cast to `const void *` or even `const unsigned char *`.